frontend/webapp/reuseable-components/input-table/index.tsx

```diff
@@ -0,0 +1,183 @@
+
+    if (!rows.length) setRows([{ ...init }]);
+  }, []);
+
+  const recordedPairs = useRef('');
+
+  useEffect(() => {
+    // Filter out rows where any values are empty
+    const validKeyValuePairs = rows.filter((row) => !Object.values(row).filter((val) => !val).length);
+    const stringified = JSON.stringify(validKeyValuePairs);
+
+    // Only trigger onChange if valid pairs have changed
+    if (recordedPairs.current !== stringified) {
+      recordedPairs.current = stringified;
+
+      if (onChange) onChange(validKeyValuePairs);
+    }
+  }, [rows, onChange]);
+
+  const handleAddRow = () => {
+    setRows((prev) => {
+      const payload = [...prev];
+      payload.push({ ...initialObject });
+      return payload;
+    });
+  };
+
+  const handleDeleteRow = (idx: number) => {
+    setRows((prev) => prev.filter((_, i) => i !== idx));
+  };
+
+  const handleChange = (key: string, val: string | number, idx: number) => {
+    setRows((prev) => {
+      const payload = [...prev];
+      payload[idx][key] = val;
+      return payload;
+    });
+  };
+
+  // Check if any key or value field is empty
+  const isAddButtonDisabled = rows.some((row) => !!Object.values(row).filter((val) => !val).length);
+
+  // adjust cell-width based on the amount of inputs on-screen,
+  // the "0.4" is to consider the delete button
+  const maxWidth = `${Math.floor(640 / (columns.length + 0.4))}px`;
+
+  return (
+    <Container>
+      <table style={{ marginBottom: '12px' }}>
+        <thead>
+          <tr>
+            {columns.map(({ title, tooltip, required }) => (
+              <th key={`input-table-head-${title}`} style={{ maxWidth }}>
+                <Tooltip text={tooltip || ''}>
+                  <HeaderWrapper>
+                    <Title>{title}</Title>
+                    {!required && (
+                      <Text color='#7A7A7A' size={14} weight={300} opacity={0.8}>
+                        (optional)
+                      </Text>
+                    )}
+                    {tooltip && <Image src='/icons/common/info.svg' alt='' width={16} height={16} style={{ marginBottom: 4 }} />}
+                  </HeaderWrapper>
+                </Tooltip>
+              </th>
+            ))}
+
+            <th>{/* this is here because of delete button */}</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {rows.map((row, idx) => (
+            <tr key={`input-table-row-${idx}`} style={{ height: '50px' }}>
+              {columns.map(({ type, keyName, placeholder }) => (
+                <td key={`input-table-row-${idx}-${keyName}`} style={{ maxWidth, padding: '0 2px' }}>
+                  <Input
+                    type={type}
+                    value={row[keyName]}
+                    onChange={({ target: { value: val } }) => handleChange(keyName, type === 'number' ? Number(val) : val, idx)}
+                    placeholder={placeholder}
+                    maxWidth={maxWidth}
+                    paddingLeft='6px'
```

Comment:
what do you think about style={{}} prop and spread it in the input?

this way you dont have to add new prop for each style and it will override the styled